frameworks/rooch-framework/sources/simple_rng.move

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+/// A simple random number generator in Move language.
+module rooch_framework::simple_rng {
+    use moveos_std::tx_context;
+    use moveos_std::timestamp;
+    use moveos_std::bcs;
+    use std::vector;
+    use std::hash;
+    use std::option;
+    use rooch_framework::transaction::{Self, TransactionSequenceInfo};
+
+    const ErrorInvalidArg: u64 = 0;
+    const ErrorInvalidU64: u64 = 1;
+    const ErrorInvalidU128: u64 = 2;
+    const ErrorInvalidSeed: u64 = 3;
+
+    fun seed(): vector<u8> {
+        // get sequence number
+        let sequence_number = tx_context::sequence_number();
+        let sequence_number_bytes = bcs::to_bytes(&sequence_number);
+
+        // get sender address
+        let sender_addr = tx_context::sender();
+        let sender_addr_bytes = bcs::to_bytes(&sender_addr);
+
+        // get now milliseconds timestamp
+        let timestamp_ms = timestamp::now_milliseconds();
+        let timestamp_ms_bytes = bcs::to_bytes(&timestamp_ms);
+
+        // construct a seed
+        let seed_bytes = vector::empty<u8>();
+        
+        // get the tx accumulator root if exists
+        let tx_sequence_info_opt = tx_context::get_attribute<TransactionSequenceInfo>();
+        if (option::is_some(&tx_sequence_info_opt)) {
+            let tx_sequence_info = option::extract(&mut tx_sequence_info_opt);
+            let tx_accumulator_root = transaction::tx_accumulator_root(&tx_sequence_info);
+            let tx_accumulator_root_bytes = bcs::to_bytes(&tx_accumulator_root);
+            vector::append(&mut seed_bytes, tx_accumulator_root_bytes);
+        } else {
+            // if it doesn't exist, get the tx hash
+            let tx_hash = tx_context::tx_hash();
+            let tx_hash_bytes = bcs::to_bytes(&tx_hash);
+            vector::append(&mut seed_bytes, tx_hash_bytes);
+        };
+
+        vector::append(&mut seed_bytes, timestamp_ms_bytes);
```

Comment:
If the user does not initiate a new transaction, considering that the global timestamp is synchronized from the BTC Block, the interval period is about 10 minutes. During this period, are the seed and random numbers obtained the same every time?

Reply:
I referred to the description of #771 and have been doing so according to that. If the timestamp and tx hash or tx accumulator root are the same during the interval period, it would be the same random number. Without additional algorithm, there is no randomness guaranteed.

Reply:
Generating a random number needs a new transaction. That transaction will include the sequencer time and update the timestamp.